diff -ru systemtap.orig/runtime/linux/print.c systemtap/runtime/linux/print.c
--- systemtap.orig/runtime/linux/print.c	2016-02-01 15:16:00.029908750 +0100
+++ systemtap/runtime/linux/print.c	2016-02-01 15:18:39.481915288 +0100
@@ -239,9 +239,9 @@
 	       THIS_MODULE->name,
 	       vstr, 
 #ifndef STAPCONF_GRSECURITY
-	       THIS_MODULE->module_core,
-	       (unsigned long) (THIS_MODULE->core_size - THIS_MODULE->core_text_size)/1024,
-               (unsigned long) (THIS_MODULE->core_text_size)/1024,
+	       THIS_MODULE->core_layout.base,
+               (unsigned long) (THIS_MODULE->core_layout.size - THIS_MODULE->core_layout.text_size)/1024,
+               (unsigned long) (THIS_MODULE->core_layout.text_size)/1024,
 #else
                THIS_MODULE->module_core_rx,
 	       (unsigned long) (THIS_MODULE->core_size_rw - THIS_MODULE->core_size_rx)/1024,
