--- a/runtime/stp_utrace.c	2017-01-23 13:55:10.607927935 +0000
+++ b/runtime/stp_utrace.c	2017-01-23 13:59:42.714865856 +0000
@@ -1078,6 +1078,19 @@
 	engine->flags = UTRACE_EVENT(QUIESCE);
 }
 
+#ifndef __set_task_state
+#ifdef CONFIG_DEBUG_ATOMIC_SLEEP
+#define __set_task_state(tsk, state_value)			\
+	do {							\
+		(tsk)->task_state_change = _THIS_IP_;		\
+		(tsk)->state = (state_value);			\
+	} while (0)
+#else
+#define __set_task_state(tsk, state_value)			\
+	do { (tsk)->state = (state_value); } while (0)
+#endif
+#endif
+
 /*
  * Get @target to stop and return true if it is already stopped now.
  * If we return false, it will make some event callback soonish.
