--- a/runtime/transport/transport.c	2016-05-30 20:56:50.941628717 +0100
+++ b/runtime/transport/transport.c	2016-05-30 20:58:03.594555618 +0100
@@ -503,7 +503,7 @@
 static inline void _stp_lock_inode(struct inode *inode)
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,16)
-	mutex_lock(&inode->i_mutex);
+	inode_lock(inode);
 #else
 	down(&inode->i_sem);
 #endif
@@ -512,7 +512,7 @@
 static inline void _stp_unlock_inode(struct inode *inode)
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,16)
-	mutex_unlock(&inode->i_mutex);
+	inode_unlock(inode);
 #else
 	up(&inode->i_sem);
 #endif
